src/main/java/org/opensearch/performanceanalyzer/decisionmaker/deciders/searchbackpressure/SearchBackPressurePolicy.java

```diff
@@ -0,0 +1,317 @@
+                }
+            }
+        }
+    }
+
+    public boolean isShardHeapThresholdTooSmall() {
+        return !searchBackPressureShardHeapIncreaseAlarm.isHealthy();
+    }
+
+    public boolean isShardHeapThresholdTooLarge() {
+        return !searchBackPressureShardHeapDecreaseAlarm.isHealthy();
+    }
+
+    public boolean isTaskHeapThresholdTooSmall() {
+        return !searchBackPressureTaskHeapIncreaseAlarm.isHealthy();
+    }
+
+    public boolean isTaskHeapThresholdTooLarge() {
+        return !searchBackPressureTaskHeapDecreaseAlarm.isHealthy();
+    }
+
+    // create alarm monitor from config
+    public SearchBpActionsAlarmMonitor createAlarmMonitor(Path persistenceBasePath) {
+        // LOG the policyConfig.getHourMonitorWindowSizeMinutes() BuketSize and dahy breanch
+        // threhsold
+        LOG.info(
+                "createAlarmMonitor with hour window: {}, bucket size: {}, hour threshold: {}, stepsize: {}",
+                policyConfig.getHourMonitorWindowSizeMinutes(),
+                policyConfig.getHourMonitorBucketSizeMinutes(),
+                policyConfig.getHourBreachThreshold(),
+                policyConfig.getSearchbpHeapStepsizeInPercentage());
+        BucketizedSlidingWindowConfig hourMonitorConfig =
+                new BucketizedSlidingWindowConfig(
+                        policyConfig.getHourMonitorWindowSizeMinutes(),
+                        policyConfig.getHourMonitorBucketSizeMinutes(),
+                        TimeUnit.MINUTES,
+                        persistenceBasePath);
+
+        // TODO: Check whether we need a persistence path to write our data
+        return new SearchBpActionsAlarmMonitor(
+                policyConfig.getHourBreachThreshold(), null, hourMonitorConfig);
+    }
+
+    // initalize all alarm monitors
+    public void initialize() {
+        LOG.info("Initializing alarms with dummy path");
+        // initialize shard level alarm for resounce unit that suggests to increase jvm threshold
+        if (searchBackPressureShardHeapIncreaseAlarm == null) {
+            searchBackPressureShardHeapIncreaseAlarm = createAlarmMonitor(SEARCHBP_DATA_FILE_PATH);
+        }
+
+        // initialize shard level alarm for resounce unit that suggests to decrease jvm threshold
+        if (searchBackPressureShardHeapDecreaseAlarm == null) {
+            searchBackPressureShardHeapDecreaseAlarm = createAlarmMonitor(SEARCHBP_DATA_FILE_PATH);
+        }
+
+        // initialize task level alarm for resounce unit that suggests to increase jvm threshold
+        if (searchBackPressureTaskHeapIncreaseAlarm == null) {
+            searchBackPressureTaskHeapIncreaseAlarm = createAlarmMonitor(SEARCHBP_DATA_FILE_PATH);
+        }
+
+        // initialize task level alarm for resounce unit that suggests to decrease jvm threhsold
+        if (searchBackPressureTaskHeapDecreaseAlarm == null) {
+            searchBackPressureTaskHeapDecreaseAlarm = createAlarmMonitor(SEARCHBP_DATA_FILE_PATH);
```

Comment:
Is there a better way to perform null check and assignment here?

Reply:
Resolved.
Added a helper function to initialize individual alarm monitor
```
    private SearchBpActionsAlarmMonitor initializeAlarmMonitor(
            SearchBpActionsAlarmMonitor alarmMonitor) {
        if (alarmMonitor == null) {
            return createAlarmMonitor(SEARCHBP_DATA_FILE_PATH);
        } else {
            return alarmMonitor;
        }
    }
```
And Call them separately inside initialize()
```
public void initialize() {
        LOG.info("Initializing alarms with dummy path");
        // initialize shard level alarm for resounce unit that suggests to increase jvm threshold
        searchBackPressureShardHeapIncreaseAlarm =
                initializeAlarmMonitor(searchBackPressureShardHeapIncreaseAlarm);

        // initialize shard level alarm for resounce unit that suggests to decrease jvm threshold
        searchBackPressureShardHeapDecreaseAlarm =
                initializeAlarmMonitor(searchBackPressureShardHeapDecreaseAlarm);

        // initialize task level alarm for resounce unit that suggests to increase jvm threshold
        searchBackPressureTaskHeapIncreaseAlarm =
                initializeAlarmMonitor(searchBackPressureTaskHeapIncreaseAlarm);

        // initialize task level alarm for resounce unit that suggests to decrease jvm threhsold
        searchBackPressureTaskHeapDecreaseAlarm =
                initializeAlarmMonitor(searchBackPressureTaskHeapDecreaseAlarm);
    }
 
```